picamera2/devices/imx500/imx500.py

```diff
@@ -45,6 +47,177 @@ class CnnOutputTensorInfoExported(ctypes.LittleEndianStructure):
         ('info', OutputTensorInfo * MAX_NUM_TENSORS)
     ]
 
+class NetworkIntrinsics:
+    def __init__(self, val=None):
+        self.__intrinsics: Optional[dict] = None
+        self.__schema = {
+            "$schema": "https://json-schema.org/draft-07/schema",
+            "title": "network_intrinsics",
+            "type": "object",
+            "properties": {
+                "task": {
+                    "type": "string",
+                    "enum": ["classification", "object detection", "segmentation"],
+                    "description": "Network task",
+                },
+                "inference_rate": {"type": "number", "minimum": 0},
+                "cpu": {
+                    "type": "object",
+                    "properties": {
+                        "bbox_normalization": {"type": "boolean"},
+                        "softmax": {"type": "boolean"},
+                        "post_processing": {"type": "string"},
+                    },
+                },
+                "input_aspect_ratio": {
+                    "type": "object",
+                    "properties": {
+                        "width": {"type": "integer", "exclusiveMinimum": 0},
+                        "height": {"type": "integer", "exclusiveMinimum": 0},
+                    },
+                    "required": ["width", "height"],
+                },
+                "classes": {
+                    "type": "object",
+                    "properties": {
+                        "labels": {"type": "array", "items": {"type": "string"}},
+                        "ignore_undefined": {"type": "boolean"},
+                    },
+                },
+            },
+        }
+        if val is not None:
+            jsonschema.validate(val, self.__schema)
+            self.__intrinsics = val
+
+    @property
+    def intrinsics(self) -> Optional[dict]:
+        return self.__intrinsics
+
+    @intrinsics.setter
+    def intrinsics(self, val):
+        jsonschema.validate(val, self.__schema)
+        self.__intrinsics = val
+
+    def __repr__(self):
+        return json.dumps(self.__intrinsics) if self.__intrinsics else ""
+
+    def __top_level_validated_insert(self, val: dict):
+        jsonschema.validate(val, self.__schema | {'additionalProperties': False})
+        self.__intrinsics = self.__intrinsics | val if self.__intrinsics else val
+
+    def __intrinsics_has_key(self, key: str) -> bool:
+        return key in self.__intrinsics if self.__intrinsics else false
+
+    def __intrinsics_get_key(self, key, default=None):
+        return self.__intrinsics.get(key, default) if self.__intrinsics else default
+
+    @property
+    def task(self) -> Optional[str]:
+        return self.__intrinsics_get_key('task')
+
+    @task.setter
+    def task(self, val: str):
+        self.__top_level_validated_insert({'task': val})
+
+    @property
+    def fps(self) -> Optional[float]:
```

Comment:
I slightly prefer ``inference_fps`` or ``inference_rate``, but no big deal.

Reply:
The property names match the previous arg names which allows us to do the hasattr/setattr trick. It's actually stored as inference_rate in the JSON.

How about changing the python property to inference_rate but then keeping an 'fps' property that aliases it?

Reply:
That works for me.